.github/workflows/enos-run.yml

```diff
@@ -12,24 +12,28 @@ on:
 
 env:
   PKG_NAME: vault
+  ARTIFACT_NAME: ${{ inputs.artifact-name }}
 
 jobs:
+  # Read Enos scenario matrix file based on artifact-name input to test
+  read-enos-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      enos-scenarios: ${{ steps.enos-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create Enos scenario matrix
+        id: enos-matrix
+        run: |
+          [[ ${{ env.ARTIFACT_NAME }} == *"ent"* ]] && echo "::set-output name=matrix::$(cat ./.github/scripts/ent.json)" || echo "::set-output name=matrix::$(cat ./.github/scripts/oss.json)"
```

Comment:
Me not being super familiar with the order of operations here, could it happen that the artifact name contains `ent` and the `cat ...ent.json` fails therefore the or clause is run and the oss scenarios are set?

Reply:
If `cat ent.json` fails the job run will error out, it won't run `oss` scenarios in that case.